### PR TITLE
Refactor trajectory

### LINKED ISSUE #143: Refactor plot_trajectory
Currently the code for trajectory chart is long is complicated and it is hard to follow what each part is reposnsible for.

Solution: extract each step into a clear, well-named function. Remove duplicate code. Make it easier to understand what is happening inside this function.

### PR DESCRIPTION
**Pull request by MonikaFu**
Closes #143 

In this PR I refactor trajectory chart plot, moving parts of the code into separate functions with clear names.

### CODE REVIEW

File: R/plot_timeline.R [hunk 1]
```diff
@@ -30,11 +30,11 @@
...
-#'                 ~line_name,           ~label, ~colour_hex,
-#'                "projected",          "Proj.",   "#1b324f",
-#'        "corporate_economy",  "Corp. Economy",   "#00c082",
-#'              "target_demo",         "Target",   "#ff9623",
-#'   "adjusted_scenario_demo",  "Adj. Scenario",   "#d0d7e1"
+#'   ~line_name, ~label, ~colour_hex,
+#'   "projected", "Proj.", "#1b324f",
+#'   "corporate_economy", "Corp. Economy", "#00c082",
+#'   "target_demo", "Target", "#ff9623",
+#'   "adjusted_scenario_demo", "Adj. Scenario", "#d0d7e1"
```
Comment (maurolepore) [inline R/plot_timeline.R:37]: Here we need to stop the styler to preserve alignment. I think the keyword is `#' styler: off` and `#' stlyer: on`.

File: R/timeline_specs.R [hunk 2]
```diff
@@ -27,11 +27,11 @@ stop_if_too_many_lines <- function(data) {
   n_colours <- nrow(r2dii_palette_colours())
   if (n_lines > n_colours) {
     msg <- sprintf(
```
Comment (maurolepore) [inline R/timeline_specs.R:29]: We can now use `glue()` but I'll do it everywhere in a separate PR.

File: tests/testthat/test-plot_timeline.R [hunk 3]
```diff
@@ -2,13 +2,13 @@ test_that("outputs the expected snapshot", {
...
-     2022,         "b",    0.6,          TRUE,
-     2023,         "b",    0.7,          TRUE,
+    ~year, ~line_name, ~value, ~extrapolated,
+    2021, "a", 0.1, FALSE,
+    2022, "a", 0.2, FALSE,
+    2023, "a", 0.3, FALSE,
+    2021, "b", 0.5, TRUE,
+    2022, "b", 0.6, TRUE,
+    2023, "b", 0.7, TRUE,
   )
```
Comment (maurolepore) [inline tests/testthat/test-plot_timeline.R:12]: We need `#' styler: off`.

**Review by maurolepore — APPROVED** [review]
LGTM. Just a few NB comments. Thanks!!!

File: R/plot_timeline.R (hunk 1 above)
Comment (MonikaFu) [reply R/plot_timeline.R:37]: I will add it as an issue as this PR is concerning a different chart.

File: tests/testthat/test-plot_timeline.R (hunk 3 above)
Comment (MonikaFu) [reply tests/testthat/test-plot_timeline.R:12]: same as above - separate PR
